### DISCUSSION TITLE
How to get the `name` value registered by `register`?

### DISCUSSION THREAD

**Discussion by chioio** [react-hook-form/react-hook-form#7077] [Q&A] [answered]
When I write a custom input component, I want to get `errors([field])` value based on the registered `name` property. But the `name` property maybe `undefined`, so I need to infer the `name` value  is  empty anywhere I needed to use it, and the `name`s value was always registered sinces component is called.
Look the code:
```ts
interface InputProps
  extends React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  > {
  label: string
}

export const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ label, ...props }, ref) => {
    const {
      getValues,
      formState: { errors },
    } = useFormContext()

    const [statusStyle, setStatusStyle] = useState<TwStyle>()

    useEffect(() => {
      if (props.name) {
        errors[props.name] &&
          setStatusStyle(tw`ring ring-red-400 border-red-500`)
        !errors[props.name] &&
          getValues(props.name) &&
          setStatusStyle(tw`ring ring-green-200 border-green-300`)
      }
    }, [props.name && getValues(props.name), props.name && errors[props.name]])

    return (
        {/* input element... */}
        {props.name && (
          <ErrorMessage
            name={props.name}
            render={({ message }) => (
              <span css={tw`absolute bottom-0 left-0 text-sm text-red-500`}>
                {message}
              </span>
            )}
          />
        )}
      </div>
    )
  },
)
```

**Comment by ritikbanger** [comment, marked as answer]
Hello @chioio, `Register` do register your inputs but the data is send only via the `handleSubmit` handler from your form. It will pass the registered inputs to a particular function you give to the `handleSubmit` handler.
Like `handleSubmit={onSubmit}` will call the function `onSubmit` on submit button.

and then we will have the onSubmit function like this:
`const onSubmit = async (data: any) => { console.log('Data: ', data)`
Now you can access the value of name by `data.name`

**Comment by ritikbanger** [comment]
@bluebill1049 Can we close this?

**Reply by bluebill1049** [reply]
yea thanks @ritikbanger
